cmd/gaia/app/sim_test.go

```diff
@@ -49,42 +50,93 @@ func init() {
 func appStateFn(r *rand.Rand, accs []simulation.Account) json.RawMessage {
 	var genesisAccounts []GenesisAccount
 
-	amt := int64(10000)
+	amount := int64(r.Intn(1e6))
+	numInitiallyBonded := int64(r.Intn(250))
+	numAccs := int64(len(accs))
+	if numInitiallyBonded > numAccs {
+		numInitiallyBonded = numAccs
+	}
+	fmt.Printf("Selected randomly generated parameters for simulated genesis: {amount: %v, numInitiallyBonded: %v}\n", amount, numInitiallyBonded)
 
 	// Randomly generate some genesis accounts
 	for _, acc := range accs {
-		coins := sdk.Coins{sdk.Coin{"steak", sdk.NewInt(amt)}}
+		coins := sdk.Coins{sdk.Coin{"steak", sdk.NewInt(amount)}}
 		genesisAccounts = append(genesisAccounts, GenesisAccount{
 			Address: acc.Address,
 			Coins:   coins,
 		})
 	}
 
-	// Default genesis state
-	govGenesis := gov.DefaultGenesisState()
-	stakeGenesis := stake.DefaultGenesisState()
-	slashingGenesis := slashing.DefaultGenesisState()
+	// Random genesis states
+	govGenesis := gov.GenesisState{
+		StartingProposalID: int64(r.Intn(100)),
```

Comment:
Would it make sense for randomized genesis states to be defined within the module itself?

Reply:
That's a good question. I think they may need to be dependent (e.g. the max evidence age for slashing should be the unbonding period), although in that case I wonder if we should just share the parameters themselves...

Reply:
If we have that sort of requirement, they definitely should be dependent when being set or be the same variable

Reply:
Lets keep things here for a bit... we lose agility by organizing things too far, and simulation is an edge application, so I think it's fine as is for now.